app/assets/javascripts/discourse/app/services/composer.js

```diff
@@ -1663,4 +1684,92 @@ export default class ComposerController extends Controller {
   clearLastValidatedAt() {
     this.set("lastValidatedAt", null);
   }
+
+  prepareFormTemplateData(event, skipValidation = false) {
```

Comment:
If possible, I'd like to have the validations live outside the composer service to keep things lean in the composer.